tick/eval/node_evaluator.go

```diff
@@ -0,0 +1,97 @@
+package eval
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/influxdata/kapacitor/tick"
+)
+
+// ErrTypeGuardFailed is returned when a speicifc value type is requested thorugh NodeEvaluator (for example: "Float64Value")
+// when the node doesn't support the given type, for example "Float64Value" is called on BoolNode
+type ErrTypeGuardFailed struct {
+	RequestedType ValueType
+	ActualType    ValueType
+}
+
+func CreateTypeGuardFailedError(requested, actual ValueType) error {
+	return &ErrTypeGuardFailed{requested, actual}
+}
+
+func (e ErrTypeGuardFailed) Error() string {
+	return fmt.Sprintf("expression returned unexpected type %v", valueTypeToName[e.ActualType])
+}
+
+type ReadOnlyScope interface {
+	Get(name string) (interface{}, error)
+}
+
+// NodeEvaluator provides a generic way for trying to fetch
+// node value, if a speicifc type is requested (so Value isn't called, the *Value is called) ErrTypeGuardFailed must be returned
+type NodeEvaluator interface {
+	EvalFloat64(scope *tick.Scope, executionState ExecutionState) (float64, error)
+	EvalInt64(scope *tick.Scope, executionState ExecutionState) (int64, error)
+	EvalString(scope *tick.Scope, executionState ExecutionState) (string, error)
+	EvalBool(scope *tick.Scope, executionState ExecutionState) (bool, error)
+	EvalRegex(scope *tick.Scope, executionState ExecutionState) (*regexp.Regexp, error)
+
+	// GetType returns the type of ValueType
+	GetType(scope ReadOnlyScope, executionState ExecutionState) (ValueType, error)
+}
+
+func createNodeEvaluator(n tick.Node) (NodeEvaluator, error) {
+	switch node := n.(type) {
+
+	case *tick.BoolNode:
+		return NodeEvaluator(&EvalBoolNode{Node: node}), nil
```

Comment:
These explicit type casts to `NodeEvaluator` are not necessary:

We should simplify them like this.
```go
	case *tick.BoolNode:
		return &EvalBoolNode{Node: node}, nil
```